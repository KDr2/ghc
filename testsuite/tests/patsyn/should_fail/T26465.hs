{-# LANGUAGE PatternSynonyms, ViewPatterns #-}

module T26465 where

f :: Eq a => a -> Maybe a
f _ = Nothing

-- Monomorphism restriction bites
-- Eq a[tau:0] => a[tau:0] -> Maybe a[tau:0]
g = f

pattern P x <- ( g -> Just x )
