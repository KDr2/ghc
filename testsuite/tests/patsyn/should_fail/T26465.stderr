T26465.hs:10:5: error: [GHC-39999]
    • Ambiguous type variable ‘a0’ arising from a use of ‘f’
      prevents the constraint ‘(Eq a0)’ from being solved.
      Relevant bindings include
        g :: a0 -> Maybe a0 (bound at T26465.hs:10:1)
      Probable fix: use a type annotation to specify what ‘a0’ should be.
      Potentially matching instances:
        instance Eq Ordering -- Defined in ‘GHC.Internal.Classes’
        instance Eq Integer -- Defined in ‘GHC.Internal.Bignum.Integer’
        ...plus 24 others
        ...plus five instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the expression: f
      In an equation for ‘g’: g = f

